### PR TITLE
[FLINK-11834] [table-planner-blink] Introduce flink logical relational nodes

### PR DESCRIPTION
**Pull request by godfreyhe**
## What is the purpose of the change

Introduce flink logical relational nodes.
This commit includes most flink logical relational nodes, the rest will be introduced by other commit.

## Brief change log

  - *Adds  nodes extended from Calcite, including `FlinkLogicalAggregate`, `FlinkLogicalCalc`,` FlinkLogicalCorrelate`, `FlinkLogicalIntersect`, `FlinkLogicalJoin`, `FlinkLogicalMinus`, `FlinkLogicalOverWindow`, `FlinkLogicalSort`, `FlinkLogicalUnion`, `FlinkLogicalValues`, `FlinkLogicalTableSourceScan`, `FlinkLogicalTableFunctionScan`*
  - *Adds new RelNode for Flink, including `Expand`, `Rank`, `Sink`, `WatermarkAssigner`*

## Verifying this change

This change is an initialization work without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/RankRange.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.plan.util
+
+sealed trait RankRange extends Serializable {
```
Comment (KurtYoung) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/RankRange.scala]: Looks like this is a part of your Rank rel node, i think should not belong to `util`

Comment (godfreyhe) [reply flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/RankRange.scala]: `RankRange` can be moved into Rank.scala file

**Comment by godfreyhe** [comment]
`FlinkLogicalTableSourceScan` is also added into this commit. The rest flink logical nodes(contains: `FlinkLogicalWindowAggregate`, `FlinkLogicalSnapshot`, `FlinkLogicalSemiJoin`, `FlinkLogicalMatch` ) will be introduced later because they depend on `Expression` or Calcite change.

**Comment by KurtYoung** [comment]
I think you can open a jira to track all unmerged logical nodes

**Comment by godfreyhe** [comment]
related issues:
https://issues.apache.org/jira/browse/FLINK-11839
https://issues.apache.org/jira/browse/FLINK-11840
https://issues.apache.org/jira/browse/FLINK-11841
https://issues.apache.org/jira/browse/FLINK-11842

**Pull request merged**